manifests/init.pp

```diff
@@ -169,4 +170,10 @@
     }
   }
 
+  if $repo_scl {
+    ::rhsm::repo { "rhel-server-rhscl-${::operatingsystemmajrelease}-rpms":
+      require => Exec['RHSM-register'],
```

Comment:
maybe the defined type rhsm::repo should take care about ordering of the resource itself instead of having each caller adding manual dependencies?

Reply:
It makes sense, but I would have no idea of how to do it, unfortunately.